exp/controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -96,7 +96,7 @@ func (s *azureManagedMachinePoolService) Reconcile(ctx context.Context) error {
 	defer done()
 
 	log.Info("reconciling managed machine pool")
-	agentPoolName := s.scope.AgentPoolSpec().Name
+	agentPoolName := s.scope.AgentPoolSpec().ResourceName()
```

Comment:
the name is just the ManagedMachinePool name so not sure we really need to get the spec here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2479/files#diff-5607bc7afbe8f18686e0e62796b8da6371e999006ddb598c5034dd7ef7040a47R146

This could just be `s.scope.Name()` where `Name()` is a function in scope that just returns `s.InfraMachinePool.Spec.Name`